archive/imo/imo1959_q1.lean

```diff
@@ -0,0 +1,32 @@
+/-
+Copyright (c) 2020 Kevin Lacker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Lacker
+-/
+
+import tactic.linarith
+
+open nat
+
+/-!
+# IMO 1959 Q1
+
+Prove that the fraction (21n+4)/(14n+3) is irreducible for every natural number n.
+
+Since Lean doesn't have a concept of "irreducible fractions" per se, we just formalize this
+as saying the numerator and denominator are relatively prime.
+-/
+
+lemma calculation (k n : ℕ) (h1 : k ∣ 21 * n + 4) (h2 : k ∣ 14 * n + 3) : k ∣ 1 :=
+have h3 : k ∣ 2 * (21 * n + 4), from dvd_mul_of_dvd_right h1 2,
+have h4 : k ∣ 3 * (14 * n + 3), from dvd_mul_of_dvd_right h2 3,
+have h5 : 2 * (21 * n + 4) + 1 = 3 * (14 * n + 3), by linarith,
+(nat.dvd_add_right h3).mp (eq.subst h5.symm h4)
+
+theorem imo1959_q1 : ∀ n : ℕ, coprime (21 * n + 4) (14 * n + 3) :=
+assume n,
+begin
```

Comment:
If you want to be super slick this whole proof can be
```suggestion
assume n, coprime_of_dvd' (calculation n)
```
by swapping the `k n` in the definition of calculation to `n k` .

Reply:
Okay